src/popup/Popup.tsx

```diff
@@ -1,23 +1,56 @@
-import React, { useEffect, useState } from 'react';
-import { getCurrentTabUrl } from '../utils';
+import React, { useCallback, useEffect, useState } from 'react';
 
 export default function Popup(): JSX.Element {
-  const [url, setUrl] = useState<string>('');
+  const [emotesEnabled, setEmotesEnabled] = useState(null);
 
   useEffect(() => {
-    getCurrentTabUrl((url) => {
-      setUrl(url || 'undefined');
+    chrome.storage.sync.get(['emotesEnabled'], (result) => {
+      setEmotesEnabled(result['emotesEnabled']);
     });
   }, []);
 
-  const sendAMessage = () => {
-    // Example of how to send a message to eventPage.ts.
-    chrome.runtime.sendMessage({ popupMounted: true });
-  };
+  const changeEmoteStatus = useCallback(() => {
+    const currentEmoteStatus = !emotesEnabled; // setState is not guaranteed to mutate state immediately, need new value
+    setEmotesEnabled(!emotesEnabled);
+    let message: string;
+    if (currentEmoteStatus) {
```

Comment:
can this can be a ternary `message = currentEmoteStatus ? 'enableEmotes' :  'disableEmotes'`?